runtime/frequency/src/lib.rs

```diff
@@ -252,28 +260,28 @@ pub struct BaseCallFilter;
 
 impl Contains<RuntimeCall> for BaseCallFilter {
 	fn contains(call: &RuntimeCall) -> bool {
-		#[cfg(not(feature = "frequency"))]
-		{
-			match call {
-				RuntimeCall::Utility(pallet_utility_call) =>
-					Self::is_utility_call_allowed(pallet_utility_call),
-				_ => true,
-			}
-		}
-		#[cfg(feature = "frequency")]
-		{
-			match call {
-				RuntimeCall::Utility(pallet_utility_call) =>
-					Self::is_utility_call_allowed(pallet_utility_call),
-				// Create provider, create application, and create schema are not allowed in mainnet for now. See propose functions.
-				RuntimeCall::Msa(pallet_msa::Call::create_provider { .. }) |
-				RuntimeCall::Msa(pallet_msa::Call::create_application { .. }) |
-				RuntimeCall::Schemas(pallet_schemas::Call::create_schema_v3 { .. }) => false,
-				#[cfg(feature = "frequency-bridging")]
-				RuntimeCall::PolkadotXcm(pallet_xcm_call) => Self::is_xcm_call_allowed(pallet_xcm_call),
-				// Everything else is allowed on Mainnet
-				_ => true,
-			}
+		match call {
+			RuntimeCall::Utility(pallet_utility_call) =>
+				Self::is_utility_call_allowed(pallet_utility_call),
+			// Block stateful-storage extrinsics if V1->V2 migration is not complete
+			// May be removed once the migration has been completed on mainnet
+			RuntimeCall::StatefulStorage(..) =>
+				pallet_stateful_storage::Pallet::<Runtime>::on_chain_storage_version() >=
+					StorageVersion::new(2),
+
+			#[cfg(feature = "frequency")]
+			// Filter out calls that are Governance actions on Mainnet
+			RuntimeCall::Msa(pallet_msa::Call::create_provider { .. }) |
+			RuntimeCall::Msa(pallet_msa::Call::create_application { .. }) |
+			RuntimeCall::Schemas(pallet_schemas::Call::create_schema_v4 { .. }) |
+			RuntimeCall::Schemas(pallet_schemas::Call::create_intent { .. }) |
+			RuntimeCall::Schemas(pallet_schemas::Call::create_intent_group { .. }) |
+			RuntimeCall::Schemas(pallet_schemas::Call::update_intent_group { .. }) => false,
+
+			#[cfg(all(feature = "frequency-bridging", feature = "frequency"))]
+			RuntimeCall::PolkadotXcm(pallet_xcm_call) => Self::is_xcm_call_allowed(pallet_xcm_call),
+			// Everything else is allowed
+			_ => true,
```

Comment:
Feature flags are weird and this help at least visually in the least
```suggestion

			// Everything else is allowed
			_ => true,
```